pkg/virt-launcher/virtwrap/nichotplug.go

```diff
@@ -87,9 +89,44 @@ func (vim *virtIOInterfaceManager) hotplugVirtioInterface(vmi *v1.VirtualMachine
 	return nil
 }
 
-func domainInterfaceFromNetwork(domain *api.Domain, network v1.Network) *api.Interface {
-	for _, iface := range domain.Spec.Devices.Interfaces {
-		if iface.Alias.GetName() == network.Name {
+func (vim *virtIOInterfaceManager) hotUnplugVirtioInterface(vmi *v1.VirtualMachineInstance, currentDomain *api.Domain) error {
+	for _, domainIface := range interfacesToHotUnplug(vmi.Spec.Domain.Devices.Interfaces, currentDomain.Spec.Devices.Interfaces) {
+		log.Log.Infof("preparing to hot-unplug %s", domainIface.Alias.GetName())
+
+		ifaceXML, err := xml.Marshal(domainIface)
+		if err != nil {
+			return err
+		}
+
+		if derr := vim.dom.DetachDeviceFlags(strings.ToLower(string(ifaceXML)), affectLiveAndConfigLibvirtFlags); derr != nil {
```

Comment:
Yeah I meant async, sorry :sweat_smile: 

Should we worry about piling detach requests to Libevirt if the operation hangs?

Reply:
I'm not sure what is pilling and where.
The API is async, so it will just return and potentially in the background QEMU will continue some processing.
Libvirt API says nothing about what will happen if we re-request the same thing if it did not finish, but I expect libvirt and QEMU to manage it somehow (e.g. fail, ignore or queue). If it does not do it well, it is a bug and they should fix it.

From our side, we are fine waiting and not blocking... we will always reconcile based on the current state.
This approach would not have worked for SR-IOV IMO, because there we need to wait for all to detach before issuing the migration command, otherwise it will fail.